packages/protocol/contracts/layer1/based/LibProposing.sol

```diff
@@ -182,7 +182,8 @@ library LibProposing {
 
         unchecked {
             if (local.params.anchorBlockId == 0) {
-                local.params.anchorBlockId = uint64(block.number - 1);
+                local.params.anchorBlockId =
+                    local.postFork ? uint64(block.number - 1) : uint64(block.number);
```

Comment:
<img width="1031" alt="image" src="https://github.com/user-attachments/assets/5039ca18-92a4-44ff-bfda-99ee7054bf54">
Why do we need to change it back here?

Reply:
changed